src/evaluate.jl

```diff
@@ -151,52 +109,57 @@ function evaluate!{T<:Number}(x::Array{Taylor1{TaylorN{T}},1}, δt::T,
     nothing
 end
 
-## Evaluates HomogeneousPolynomials and TaylorN on a val of the nv variable
-## using Horner's rule on the nv variable
-function horner{T<:Number,S<:NumberNotSeriesN}(a::HomogeneousPolynomial{T},
-        b::Tuple{Int,S} )
-    nv, val = b
-    numVars = get_numvars()
-    @assert 1 <= nv <= numVars
+"""
+    evaluate(a, vals)
+
+Evaluate a `HomogeneousPolynomial` polynomial at `vals`.
+"""
+function evaluate{T<:Number,S<:TaylorSeries.NumberNotSeriesN}(a::HomogeneousPolynomial{T},
+        vals::Array{S,1} )
+    @assert length(vals) == get_numvars()
+
+    num_vars = get_numvars()
+    ct = TaylorSeries.coeff_table[a.order+1]
     R = promote_type(T,S)
-    @inbounds indTb = coeff_table[a.order+1]
-    suma = TaylorN(zero(R), a.order)
-
-    # Horner's rule on the nv variable
-    for ord = a.order : -1 : 0
-        suma_ord = TaylorN(zero(R), a.order)
-        posOrd = order_posTb(a.order, nv, ord)
-        neworder = a.order-ord
-        for pos in posOrd
-            c = a[pos]
-            iIndices = copy(indTb[pos])
-            iIndices[nv] = 0
-            kdic = in_base(get_order(), iIndices)
-            newpos = pos_table[neworder+1][kdic]
-            zhp = HomogeneousPolynomial([zero(R)], neworder)
-            zhp[newpos] = a[pos]
-            suma_ord += TaylorN(zhp, a.order)
-        end
-        if ord == a.order
-            suma += suma_ord
-        else
-            suma = suma*val + suma_ord
+    suma = zero(R)
+
+    for (i,a_coeffs) in enumerate(a.coeffs)
+        tmp = vals[1]^(ct[i][1])
```

Comment:
To answer @dpsanders, if I understand correctly the code, what you are doing is, for each coefficient of `a::HomoegeneousPolynomial{T}` you are computing the actual monomial term which means to know the actual powers in the different variables, and perform the actual product. The reason that this outperforms the old code, as you say, is because you are not computing `HomoegeneousPolynomial{T}` quantities (i.e., you are not allocating any vector), but only `T` (or the correctly promoted type). Nice!

Part of the reason the code was writen as it was, was to be accurate as possible. Here we are loosing some accuracy. Do you have any ideas on this? Perhaps if you calculate each term, save it in a vector, and them add them up from the smallest to the biggest we can get a bit more precision at the cost of time, though.